realm/realm-library/src/androidTest/kotlin/io/realm/entities/DictionaryClass.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.entities
+
+import io.realm.Mixed
+import io.realm.RealmDictionary
+import io.realm.RealmList
+import io.realm.RealmObject
+
+open class DictionaryClass : RealmObject() {
+
+    var myMixedDictionary: RealmDictionary<Mixed>? = null
```

Comment:
Shouldn't also allow this to be marked as `val`? I don't think we do this for RealmList right now` but it would make a lot of sense, since that is what the semantics are.

Reply:
Just tried `val immutableList: RealmList<Boolean> = RealmList()` and it compiles fine. I will add it to dictionaries too.

Reply:
@cmelchior any idea how to test this from the annotation processor perspective? We can't use Kotlin source files in the processor tests.

Reply:
Use the Java construct that matches what Kotlin compiles to in bytecode. In this case `public final RealmDictionary<Mixed> = new RealmDictionary()`

Reply:
🤦 just had a brain meltdown...